python/ray/air/tests/execution/test_e2e_tune_flow.py

```diff
@@ -0,0 +1,207 @@
+from collections import defaultdict
```

Comment:
This is a comprehensive test, you may want to read `test_tracked_actor.py` and `test_tracked_actor_task.py` first.

In this test, I built a "fake-tune" flow. It basically does the same thing as Ray Tune does: Request more actors than can fit on the cluster, run one function on the actors for a number of iterations, and stop the actor afterwards. If anything fails, restart the actor and retry the trial.

With this test passing, this basically shows that we can easily use the new abstractions to implement Ray Tune with the new manager.

I can implement a similar flow for a Ray Train-like loop.

Reply:
this is very nice!